plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotMetadata.java

```
@@ -187,11 +177,11 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
     @Override
     public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> schemaNameOrNull)
     {
-        ImmutableList.Builder<SchemaTableName> builder = ImmutableList.builder();
-        for (String table : getPinotTableNames()) {
+        ImmutableSet.Builder<SchemaTableName> builder = ImmutableSet.builder();
```

Won't this silently swallow colliding names?